cocos/scripting/lua-bindings/manual/Cocos2dxLuaLoader.cpp

```diff
@@ -77,33 +73,55 @@ extern "C"
         {
             if (next == std::string::npos)
                 next = searchpath.length();
-            std::string prefix = searchpath.substr(begin, next);
+            std::string prefix = searchpath.substr(begin, next-begin);
             if (prefix[0] == '.' && prefix[1] == '/')
-            {
                 prefix = prefix.substr(2);
-            }
 
-            pos = prefix.find("?.lua");
-            chunkName = prefix.substr(0, pos) + filename + BYTECODE_FILE_EXT;
-            if (utils->isFileExist(chunkName))
+            pos = prefix.rfind(BYTECODE_FILE_EXT);
+            if (pos != std::string::npos && pos == prefix.length() - BYTECODE_FILE_EXT.length())
+            {
+                prefix = prefix.substr(0, pos);
+            }
+            else
+            {
+                pos = prefix.rfind(NOT_BYTECODE_FILE_EXT);
+                if (pos != std::string::npos && pos == prefix.length() - NOT_BYTECODE_FILE_EXT.length())
+                    prefix = prefix.substr(0, pos);
+            }
+            pos = prefix.find_first_of("?", 0);
+            while (pos != std::string::npos)
+            {
+                prefix.replace(pos, 1, filename);
```

Comment:
It is lua's standart, refer to https://www.lua.org/manual/5.1/manual.html#pdf-package.loaders, in case of `./lib/?/init.lua` we should try `./lib/filename/init.lua`.